poc/tenant-controller/pkg/controllers/tenants/controller.go

```diff
@@ -63,7 +70,21 @@ func (c *Controller) Run(ctx context.Context) error {
 func (c *Controller) createTenant(obj *tenantsapi.Tenant) {
 	glog.Info("createTenant: %#v", obj)
 	syncRBACForTenantCRD(obj)
-	// TODO create namespace
+
+	// TODO Add later ... sanity checks to ensure namespaces being requested are valid and not already assigned to another tenant
+	c.tenantsList = make(map[string]tenantsapi.Tenant)
+	c.tenantsList[obj.ObjectMeta.Name] = *obj
+
+	namespace := corev1.Namespace{}
+
+	for n, _ := range obj.Spec.Namespaces {
+		namespace.ObjectMeta.Name = obj.Spec.Namespaces[n].Name
+		//Create namespace
+		c.k8sclient.CoreV1().Namespaces().Create(&namespace)
```

Comment:
We will need to handle errors on Create. I suggest a TODO comment here.

Reply:
Sure, will add.